### PR TITLE
Make http_resp_hdr_len and http_resp_size configurable

### PR DESCRIPTION
**Pull request by dasrecht**
Make http_resp_hdr_len and http_resp_size configurable for large header sizes. This can happen if we use varnish to cache sites that use a huge amount of cache tags

### CODE REVIEW

File: images/varnish/Dockerfile [hunk 1]
```diff
@@ -2,7 +2,6 @@ ARG IMAGE_REPO
 FROM ${IMAGE_REPO:-lagoon}/commons as commons
 FROM alpine
 
-MAINTAINER amazee.io
```
Comment (Schnitzel) [inline images/varnish/Dockerfile]: why is this removed?

File: images/varnish/Dockerfile [hunk 2]
```diff
@@ -31,5 +30,8 @@ EXPOSE 8080
 # tells the local development environment on which port we are running
 ENV LAGOON_LOCALDEV_HTTP_PORT=8080
 
+# Make http_resp_hdr_len and http_resp_size configurable
+ENV HTTP_RESP_HDR_LEN=${http_resp_hdr_len:-8k}
```
Comment (Schnitzel) [inline images/varnish/Dockerfile]: this will not allow to overwrite the variables in runtime, but instead in Buildtime which I think it would be good to happen in Runtime

Comment (Schnitzel) [reply images/varnish/Dockerfile]: or actually it does allow to overwrite the variables, but then we don't need the `${http_resp_hdr_len:-8k}` in there, and just use `HTTP_RESP_HDR_LEN=8k`

**Review by Schnitzel — CHANGES_REQUESTED** [review]

File: images/varnish/Dockerfile (hunk 1 above)
Comment (dasrecht) [reply images/varnish/Dockerfile]: Because the MAINTAINER flags are deprected. We can add it via a label. Can create a seperate PR for that and go trough our images.


https://docs.docker.com/engine/reference/builder/#maintainer-deprecated

Comment (dasrecht) [reply images/varnish/Dockerfile]: i'll reverted that change and we can do that in a seperate pr.

File: images/varnish/Dockerfile (hunk 2 above)
Comment (dasrecht) [reply images/varnish/Dockerfile]: updated :)

File: images/varnish/docker-entrypoint [hunk 3]
```diff
@@ -2,4 +2,4 @@
 
 ep /etc/varnish/*
 
-exec "$@"
+exec /usr/sbin/varnishd -a :8080 -T :6082 -F -f /etc/varnish/default.vcl -S /etc/varnish/secret -p http_resp_hdr_len=$HTTP_RESP_HDR_LEN -p http_resp_size=$HTTP_RESP_SIZE
```
Comment (Schnitzel) [inline images/varnish/docker-entrypoint:5]: we can't have that inside an entrypoint, or we will never be able to start a container with `docker run --rm -it lagoon/varnish sh`

so we need to create it's own shell script that is then used in `CMD`

**Review by Schnitzel — CHANGES_REQUESTED** [review]

File: images/varnish/docker-entrypoint (hunk 3 above)
Comment (dasrecht) [reply images/varnish/docker-entrypoint:5]: updated and fixed a few smaller things on the way there

**Comment by dasrecht** [comment]
All checks are green :)

**Review by Schnitzel — APPROVED** [review]
